app/models/miq_provision_virt_workflow.rb

```diff
@@ -303,11 +302,13 @@ def update_field_visibility_linked_clone(_options = {}, f)
     end
   end
 
-  def update_field_required
-    fields(:service) do |fn, f, _dn, _d|
-      f[:required] =  supports_pxe? ? f[:required] : false if [:pxe_image_id, :pxe_server_id].include?(fn)
-      f[:required] =  supports_iso? ? f[:required] : false if [:iso_image_id].include?(fn)
-    end
+  def update_field_required(f)
+    show_flag = self.supports_pxe? ? :edit : :hide
+    f[show_flag] += [:pxe_image_id, :pxe_server_id]
+
+    show_flag = self.supports_iso? ? :edit : :hide
+    f[show_flag] += [:iso_image_id]
+    f
```

Comment:
No need to return object here, the caller will see the updates in the object they passed.